org.eclipse.xtext.ui.shared/src/org/eclipse/xtext/ui/shared/internal/ListenerRegistrar.java

```diff
@@ -41,9 +46,25 @@ public void initialize() {
 		// after the fact, the project-open event is already done. Thus no build is triggered for the newly
 		// opened project and the index state is corrupt.
 		// thus we trigger a recovery build here
-		if (builderState.isEmpty()) {
-			buildManager.scheduleBuildIfNecessary(Arrays.asList(workspace.getRoot().getProjects()), IBuildFlag.RECOVERY_BUILD);
+		class RecoveryBuildTrigger extends WorkspaceJob {
```

Comment:
@szarnekow what do you think about making this class public?

Reply:
(an thus move outside the method) or do we think its ok to c&p here in case of customization?